packages/runtime/src/templates/getApiHandler.ts

```diff
@@ -48,6 +47,8 @@ const makeApiHandler = ({ conf, app, pageRoot, page, NextServer }: MakeApiHandle
     require.resolve('./pages.js')
   } catch {}
 
+  const NetlifyNextServer: NetlifyNextServerType = getNetlifyNextServer(NextServer)
```

Comment:
commit messages mention trial and error process when this was added here - is this actually needed in api handler or just leftover from trials?

If it's needed - (not for this PR) I do wonder wether we should try to merge `getApiHandler` and `getHandler` into one as they become very similar now (not exactly the same) in future to make it easier to maintain going forward

Reply:
this is needed - the underlying problem was that if the plain `NextServer` impl is used, then `revalidate` fails because we're overriding the revalidate impl in `NetlifyNextServer`. In the past, `getApiHandler` was *only* used for scheduled functions + background functions, which won't use `revalidate`, so this wasn't a problem.

Agree that we should try to merge the two going forward. I'll open an issue to track that.